text/0000-promote-aarch64-unknown-linux-gnu-to-tier1.md

```diff
@@ -0,0 +1,188 @@
+
+ - That said, Arm AArch64 silicon is either already prevalent or is en-route to prevalance in a wide spectrum of application domains ranging from 'traditional' embedded systems at one end of the spectrum, on to mobile phones, clam-shell devices, desktops, vehicle autonomy controllers, datacenter servers etc all the way to high performance super-computers.
+
+ - The evidence to that effect is too numerous to quote but generally easy to verify openly. 
+
+ - It is fair to state that this is an ongoing reality which is unlikely to stop trending upwards and sidewards for the forseeable future.
+
+ - Software stacks built for those domains predominantly use an AArch64 Linux kernel build.
+
+ - Rust presents an attractive value proposition across all such domains, irrespective of the underlying processor architecture.
+
+ - **As such, the Rust aarch64-unknown-linux-gnu target's ecosystem presents very strong viability for the long term.**
+
+**2.b. The long term viability of supporting the target should be clear.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - It is worth calling out, in the same vein as the previous point, that given the increasing prevalance of AArch64 silicon deployments and given Rust's general value proposition, **supporting the Rust aarch64-unknown-linux-gnu target presents very strong viability for the long term.**
+
+**2.c. The target must have substantial and widespread interest within the Rust developer community.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - It is generally fair to state that **there is already substantial and widespread interest for the aarch64-unknown-linux-gnu target in the Rust developer commmunity**.
+
+ - It is also generally fair to state that there is a clear upward trend in the use of AArch64 systems as self hosted development environments. 
+
+ - Most major operating system environments support hosted development on AArch64 based systems and this trend is increasing.
+
+ - As a somewhat related note: Slow but steady progress is being made to support Windows AArch64 targets, initially for cross-platform development. This shall inevitably trend towards hosted development.
+
+ - As such, **it is very likely that developer interest in Rust on aarch64-unknown-linux-gnu will continue to increase in the medium to long term.**
+
+**2.d. The target must serve the interests of multiple production users of Rust across multiple organizations or projects.**
+
+ - It is hard to concretely quantify this aspect.
+
+ - Most major Arm software ecosystem partners are either already using Rust extensively, or are building up to extensive use. A few publically known examples are Microsoft and Google. There are many more.
+
+ - Arm itself recognises Rust as an important component to consider in a broader horizontal safety and security foundation across multiple processor portfolios. 
+
+ - Arm has dedicated a small team to help improve Rust for the aarch64-unknown-linux-gnu target.
```

Comment:
Who is on this team? Do they have a track record of contribution to Rust and LLVM? Don't necessarily need to list names here, but some kind of reference to an official group might be good.

Reply:
Fair point. 

Me and the team have been put on the #t-compiler/arm Zulip channel by the core team. I’ll add a reference. 

Contributions have begun but it’s early days to be fair. All of us are embedded systems folks and Rust aficionados.

Given the official mandate - which is to work with the community towards improving Rust on AArch64 - contributions will tick upwards. We’ve a number of streams in flight to do with SIMD intrinsics - all in consultation with the core team. 

As a related aside: A key value add here is to act as a bi-directional channel to more specialised groups inside Arm working on processor architecture, toolchains etc. 

The intent is to get Rust to pick up on the continuous innovations in the Arm architecture. Stuff like memory tagging extensions, capability architecture etc but I digress.

Reply:
> Me and the team have been put on the #t-compiler/arm Zulip channel by the core team. I’ll add a reference.

Probably we could also mention that there's a marker team for arm: https://github.com/rust-lang/team/blob/master/teams/arm.toml

Reply:
@raw-bin Hi, I've begun to add additional ARM intrinsics. I've even got a little [tracker sheet](https://docs.google.com/spreadsheets/d/1MqW1g8c7tlhdRWQixgdWvR4uJHNZzCYAf4V0oHjZkwA/edit#gid=0) that I'm keeping updated to help myself stay organized. If you've got a team also working on it that's great (I don't think I could do it all myself XD), but I hope yall can do a series of smaller PRs to `stdarch` rather than one huge PR our of nowhere, to avoid duplicating each other's work.

Reply:
Hey @Lokathor: Thanks for sharing! Yes we should sync up! I'll fix something up with you and the team ASAP!

Reply:
I've added commits to address comments from @JohnTitor and @lambda.

I'm setting up a sync with @Lokathor.

I've responded hopefully suitably to the other good comments.

I'm now going to resolve this conversation. Happy to engage further if needed.